tests/bluetooth/tester/src/btp_aics.c

```diff
@@ -0,0 +1,608 @@
+	tester_set_bit(rp->data, BTP_AICS_MUTE_DISABLE);
+	tester_set_bit(rp->data, BTP_AICS_GAIN_SETTING_PROP);
+	tester_set_bit(rp->data, BTP_AICS_TYPE);
+	tester_set_bit(rp->data, BTP_AICS_STATUS);
+	tester_set_bit(rp->data, BTP_AICS_STATE);
+
+	/* octet 2 */
+	tester_set_bit(rp->data, BTP_AICS_DESCRIPTION);
+
+	*rsp_len = sizeof(*rp) + 2;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+void btp_send_aics_state_ev(struct bt_conn *conn, int8_t gain, uint8_t mute, uint8_t mode)
+{
+	struct btp_aics_state_ev ev;
+	struct bt_conn_info info;
+
+	(void)bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.gain = gain;
+	ev.mute = mute;
+	ev.mode = mode;
+
+	tester_event(BTP_SERVICE_ID_AICS, BTP_AICS_STATE_EV, &ev, sizeof(ev));
+}
+
+void btp_send_gain_setting_properties_ev(struct bt_conn *conn, uint8_t units, int8_t minimum,
+					 int8_t maximum)
+{
+	struct btp_gain_setting_properties_ev ev;
+	struct bt_conn_info info;
+
+	(void)bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.units = units;
+	ev.minimum = minimum;
+	ev.maximum = maximum;
+
+	tester_event(BTP_SERVICE_ID_AICS, BTP_GAIN_SETTING_PROPERTIES_EV, &ev, sizeof(ev));
+}
+
+void btp_send_aics_input_type_event(struct bt_conn *conn, uint8_t input_type)
+{
+	struct btp_aics_input_type_ev ev;
+	struct bt_conn_info info;
+
+	(void)bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.input_type = input_type;
+
+	tester_event(BTP_SERVICE_ID_AICS, BTP_AICS_INPUT_TYPE_EV, &ev, sizeof(ev));
+}
+
+void btp_aics_status_ev(struct bt_conn *conn, bool active)
+{
+	struct btp_aics_status_ev ev;
+	struct bt_conn_info info;
+
+	(void)bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.active = active;
+
+	tester_event(BTP_SERVICE_ID_AICS, BTP_AICS_STATUS_EV, &ev, sizeof(ev));
+}
+
+void btp_aics_description_ev(struct bt_conn *conn, uint8_t data_len, char *description)
+{
+	struct btp_aics_description_ev *ev;
+	struct bt_conn_info info;
+
+	(void)bt_conn_get_info(conn, &info);
+
+	net_buf_simple_init(rx_ev_buf, 0);
+
+	ev = net_buf_simple_add(rx_ev_buf, sizeof(*ev));
+
+	bt_addr_le_copy(&ev->address, info.le.dst);
+
+	ev->data_len = data_len;
+	memcpy(ev->data, description, data_len);
+
+	tester_event(BTP_SERVICE_ID_AICS, BTP_AICS_DESCRIPTION_EV, ev, sizeof(*ev) + data_len);
+}
+
+static uint8_t aics_set_gain(const void *cmd, uint16_t cmd_len,
+			     void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_aics_set_gain_cmd *cp = cmd;
+
+	LOG_DBG("AICS set gain %d", cp->gain);
+
+	if (!bt_addr_le_eq(&cp->address, BT_ADDR_LE_ANY)) {
+		for (uint8_t i = 0; i < aics_client_instance.aics_cnt; i++) {
+			if (bt_aics_gain_set(aics_client_instance.aics[i], cp->gain) != 0) {
+				return BTP_STATUS_FAILED;
+			}
+		}
+	} else {
+		for (uint8_t i = 0; i < aics_server_instance.aics_cnt; i++) {
+			if (bt_aics_gain_set(aics_server_instance.aics[i], cp->gain) != 0) {
+				return BTP_STATUS_FAILED;
+			}
+		}
+	}
```

Comment:
The two loops are identical - Is that intentional? If so, then the `if` makes no difference

Reply:
Notice that I iterate over aics client instances in the first one and aics server instances in second one. Thought this might be good solution when a function (e.g. aics_mute) might be used either by server or client. I could distinguish them and make separate btp commands for client (with address) and server (w/o address), but then there would be so many of them. So in this case, when this btp command is called with BT_ADDR_LE_ANY passed as an argument, the operation is executed on server instance.

Reply:
Ah, yes, missed that small difference. Thanks for pointing that out. 

One comment on this though (aside from that this perhaps should have been 2 different functions), is that in `micp_mic_ctlr_discover_cb` you only ever care about a single AICS instance (and similarly we also only support 1 in the Kconfig options). 

If we want to support all or just 1 instances for everything, then we should be consistent about that.

Reply:
Right... Since there are no MICP tests that include more than one AICS instance, I will delete the iteration over client instances for now. For server case, I'll need to keep the loop beacause VCP Volume Renderer tests include two AICS instances, but it's not specified on which instance we should perform AICS operation.